src/main/java/org/iot/dsa/dslink/dynamodb/DynamoDBNode.java

```diff
@@ -0,0 +1,469 @@
+        declareDefault(Constants.SCANDYNAMODB, makeScanDynamoDBAction());
+        declareDefault(Constants.PUTITEMDYNAMODB, makePutItemDynamoDBAction());
+        declareDefault(Constants.BATCHPUTITEMSDYNAMODB, makeBatchPutItemDynamoDBAction());
+        declareDefault(Constants.UPDATEITEMDYNAMODB, makeUpdateItemDynamoDBAction());
+        declareDefault(Constants.DELETEITEMDYNAMODB, makeDeleteItemDynamoDBAction());
+    }
+
+    @Override
+    protected void onStarted() {
+        super.onStarted();
+    }
+
+    /**
+     * Override this method and Initializes nodes and parameters once Node is Stable.
+     */
+    @Override
+    protected void onStable() {
+        init();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Private Methods
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Initializes DynamoDB Connection Node
+     */
+    private void init(){
+        // Call this here in init which is called in onStable.
+        // This is needed because parameter is not available at that time
+        if (!canConnect()) {
+            return;
+        }
+        try {
+            client = new DynamoDBDSAClient(getAccessKey(),getSecretKey(),getRegion(),getEndpoint());
+            client.listtable();
+            connOk();
+        } catch (Exception e) {
+            configFault(e.getMessage().toString());
+            throw new IllegalStateException(e.getMessage().toString());
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Make Action Methods
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Make Edit Settings Action
+     */
+    private DSAction makeEditDynamoDBAction() {
+        DSAction act = new DSAction.Parameterless() {
+            @Override
+            public ActionResult invoke(DSInfo info, ActionInvocation invocation) {
+                ((DynamoDBNode) info.get()).editDynamoDB(invocation.getParameters());
+                return null;
+            }
+        };
+        act.addParameter(Constants.ACCESSKEY, DSString.valueOf(getAccessKey()), "AWS Access Key");
+        act.addParameter(Constants.SECRETKEY, DSString.valueOf(getSecretKey()), "AWS Secret Key");
+        act.addParameter(Constants.REGION, DSString.valueOf(getRegion()), "Region");
+        act.addParameter(Constants.ENDPOINT, DSString.valueOf(getEndpoint()), "End Point");
+        return act;
+    }
+
+    private DSAction makeQueryDynamoDBAction() {
+        DSAction act = new DSAction.Parameterless() {
+            @Override
+            public ActionResult invoke(DSInfo info, ActionInvocation invocation) {
+                return ((DynamoDBNode) info.get()).queryDynamoDB(this,invocation.getParameters());
+
+            }
+        };
+        act.addParameter(Constants.TABLENAME, DSValueType.STRING, "Table Name");
```

Comment:
Would it be possible to have a drop-down for Table Name?

Reply:
We will chec it.

Reply:
Done!